modin/core/execution/ray/implementations/pandas_on_ray/partitioning/partition.py

```diff
@@ -377,3 +375,53 @@ def _get_index_and_columns(df):  # pragma: no cover
         The number of columns.
     """
     return len(df.index), len(df.columns)
+
+
+PandasOnRayDataframePartition._eager_exec_func = PandasOnRayDataframePartition.apply
+PandasOnRayDataframePartition._lazy_exec_func = (
+    PandasOnRayDataframePartition.add_to_apply_calls
+)
+
+
+def _set_lazy_exec_mode(mode, put_func=LazyExecution.put):
+    """
+    Configure lazy execution mode for PandasOnRayDataframePartition.
+
+    Parameters
+    ----------
+    mode : str
+    put_func : Callable, default: LazyExecution.put
+        The original reference to the LazyExecution.put function.
+    """
+    get_logger().debug(f"Ray lazy execution mode: {mode}")
+    if put_func is not None:
+        put_func(mode)
+    if mode == "auto":
+        PandasOnRayDataframePartition.apply = (
+            PandasOnRayDataframePartition._eager_exec_func
+        )
+        PandasOnRayDataframePartition.add_to_apply_calls = (
+            PandasOnRayDataframePartition._lazy_exec_func
+        )
+    elif mode == "on":
+
+        def lazy_exec(self, func, *args, **kwargs):
+            self._lazy_exec_func(func, *args, length=None, width=None, **kwargs)
+
+        PandasOnRayDataframePartition.apply = lazy_exec
+        PandasOnRayDataframePartition.add_to_apply_calls = (
+            PandasOnRayDataframePartition._lazy_exec_func
+        )
+    elif mode == "off":
+
+        def eager_exec(self, func, *args, length=None, width=None, **kwargs):
+            self._eager_exec_func(func, *args, **kwargs)
+
+        PandasOnRayDataframePartition.apply = (
+            PandasOnRayDataframePartition._eager_exec_func
+        )
+        PandasOnRayDataframePartition.add_to_apply_calls = eager_exec
+
+
+_set_lazy_exec_mode(LazyExecution.get(), None)
+LazyExecution.put = _set_lazy_exec_mode
```

Comment:
I would better use the [subscribe interface](https://github.com/modin-project/modin/blob/5f6f80942c53012a3e0e85c0479b3cc4f56253cf/modin/config/pubsub.py#L283)
```suggestion
LazyExecution.subscribe(_set_lazy_exec_mode)
```